docusaurus/docs/iOS/guides/working-with-messagelist-layout.md

```diff
@@ -0,0 +1,225 @@
+- `bubbleThreadMetaContainer` is a vertical container which holds `bubbleView` at the top and `metadataContainer` at the bottom by default. You can easily switch the positions for those elements or even add your own according to your needs.
+- `metadataContainer` is a horizontal container which holds  `authorNameLabel` , `timestampLabel` and `onlyVisibleForYouLabel`. 
+- `bubbleView`  is view that embeds inside `bubbleContentContainer` which is responsible for displaying `quotedMessageView` and `textView`
+
+
+:::danger `bubbleView` vs `bubbleContentContainer`
+ When `ChatMessageContentView`'s `options` contain `.bubble` option, the `bubbleView` is added to `bubbleThreadMetaContainer`. If the option is not contained, the hierarchy contains only `bubbleContentContainer` as subview of `bubbleThreadMetaContainer`
+:::
+
+### Let's achieve this layout: 
+
+ ![](../assets/messagelist-layout-custom.png)
+
+With knowledge from few lines above, it implicates that we need to subclass `ChatMessageContentView` and only switch the `metadataContainer` with `bubbleView`/`bubbleThreadContainer`. 
+
+So let's start: 
+
+First we need to delete the bubble from `layoutOptionsResolver`
+```swift
+
+final class CustomMessageOptionsResolver: ChatMessageLayoutOptionsResolver {
+    override func optionsForMessage(
+        at indexPath: IndexPath,
+        in channel: ChatChannel,
+        with messages: AnyRandomAccessCollection<ChatMessage>
+    ) -> ChatMessageLayoutOptions {
+        // First let's get the default options for the message and clean them up.
+        var options = super.optionsForMessage(at: indexPath, in: channel, with: messages)
+        options.remove([.flipped, .bubble, .timestamp, .avatar, .avatarSizePadding, .authorName, .threadInfo, .reactions])
+
+        // Little helper to figure out if the message is first in group.
+        let isFirstInGroup: Bool = {
+            let messageIndex = messages.index(messages.startIndex, offsetBy: indexPath.item)
+            let message = messages[messageIndex]
+            guard messageIndex < messages.index(before: messages.endIndex) else { return true }
+            let previousMessage = messages[messages.index(after: messageIndex)]
+            guard previousMessage.author == message.author else { return true }
+            let delay = previousMessage.createdAt.timeIntervalSince(message.createdAt)
+            return delay > minTimeIntervalBetweenMessagesInGroup
+        }()
+
+        // If the message is first in group and sent by someone, let's add metadata to it.
+        if isFirstInGroup {
+            options.insert([.avatar, .timestamp, .authorName])
+        } else {
+            // The message needs to have avatarView-sized padding otherwise the message below the first one would 
+            // have leading on the level of the avatar displayed in first message.
+            options.insert(.avatarSizePadding)
+        }
+        return options
+    }
+}
+```
```

Comment:
You can simplify this by assuming we want a slack-like layout (no grouping)

Reply:
hmm this actually is taken from Slack clone demo, where it has grouping. I think I should keep it in here, otherwise it looks like this... you can suggest some change if you'd like... 


<img width="456" alt="Snímek obrazovky 2021-06-16 v 15 37 46" src="https://user-images.githubusercontent.com/17381941/122229164-ce935b00-ceb8-11eb-861c-ae9811608ad8.png">

Reply:
I think it looks better without grouping

Reply:
Done.